src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.operation.BalanceOperation;
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.operation.PurchaseOperation;
+import core.basesyntax.operation.ReturnOperation;
+import core.basesyntax.operation.SupplyOperation;
+import core.basesyntax.service.DataParser;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.serviceimpl.DataParserImpl;
+import core.basesyntax.serviceimpl.FileReaderImpl;
+import core.basesyntax.serviceimpl.FileWriterImpl;
+import core.basesyntax.serviceimpl.ReportGeneratorImpl;
+import core.basesyntax.serviceimpl.ShopServiceImpl;
+import core.basesyntax.strategy.Strategy;
+import core.basesyntax.strategy.StrategyImpl;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String FILE_FROM = "src/main/resources/input.csv";
+    private static final String FILE_TO = "src/main/resources/output.csv";
+
+    public static void main(String[] args) {
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.readFromFile(FILE_FROM);
+
+        Map<Operation, OperationHandler> operationMap = Map.of(
+                Operation.BALANCE, new BalanceOperation(),
+                Operation.SUPPLY, new SupplyOperation(),
+                Operation.PURCHASE, new PurchaseOperation(),
+                Operation.RETURN, new ReturnOperation());

```

Comment:
The creation of the operation handlers and their association with operations should not be done directly in the `main()` method. Consider moving this logic into a factory or a configuration class that can be easily modified or tested independently.